dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -230,6 +230,68 @@ object TypedEncoder {
       )
   }
 
+  implicit val timeInstant: TypedEncoder[java.time.Instant] = new TypedEncoder[java.time.Instant] {
+    def nullable: Boolean = false
+
+    def jvmRepr: DataType = ScalaReflection.dataTypeFor[java.time.Instant]
+    def catalystRepr: DataType = TimestampType
+
+    def toCatalyst(path: Expression): Expression =
+      StaticInvoke(
+        DateTimeUtils.getClass,
+        TimestampType,
+        "instantToMicros",
+        path :: Nil,
+        returnNullable = false)
+
+    def fromCatalyst(path: Expression): Expression =
+      StaticInvoke(
+        staticObject = DateTimeUtils.getClass,
+        dataType = jvmRepr,
+        functionName = "microsToInstant",
+        arguments = path :: Nil,
+        propagateNull = true
+      )
+  }
+
+  implicit val timeDuration: TypedEncoder[java.time.Duration] = new TypedEncoder[java.time.Duration] {
+    def nullable: Boolean = false
+
+    def jvmRepr: DataType = ScalaReflection.dataTypeFor[java.time.Duration]
```

Comment:
How about using `FramelessInternals.objectTypeFor` here to aling with other encoders implementations?

Reply:
Ok, I have changed duration and period jvmRepr